clang/lib/3C/ConstraintVariables.cpp

```diff
@@ -1298,7 +1298,7 @@ bool PointerVariableConstraint::solutionEqualTo(
         auto J = OthCVars.begin();
         // Special handling for zero width arrays so they can compare equal to
         // ARR or PTR.
-        if (IsZeroWidthArray) {
+        if (ComparePtyp && IsZeroWidthArray) {
```

Comment:
Can you explain what this is needed for? It would be nice to see a comment for the method `solutionEqualTo` that says why you would or would not set this flag.